yt/data_objects/static_output.py

```diff
@@ -1244,6 +1245,87 @@ def add_field(self, name, function=None, sampling_type=None, **kwargs):
         deps, _ = self.field_info.check_derived_fields([name])
         self.field_dependencies.update(deps)
 
+    def add_deposited_mesh_field(self, deposit_field, ptype='all'):
+        """Add a new deposited mesh field
+
+        Creates a new particle field which has the value of the
+        *deposit_field* at the location of each particle of type
+        *ptype*.
+
+        Parameters
+        ----------
+
+        deposit_field : tuple
+           The field name tuple of the mesh field to be deposited onto
+           the particles. This must be a field name tuple so yt can
+           appropriately infer the correct particle type.
+        ptype : string, default 'all'
+           The particle type onto which the deposition will occur.
+
+        Returns
+        -------
+
+        The field name tuple for the newly created field.
+
+        Examples
+        --------
+        >>> ds = yt.load('output_00080/info_00080.txt')
+        ... ds.add_deposited_mesh_field(('gas', 'density'), ptype='all')
+
+        >>> print('The density at the location of the particle is:')
+        ... print(ds.r['all', 'cell_gas_density'])
+        The density at the location of the particle is:
+        [9.33886124e-30 1.22174333e-28 1.20402333e-28 ... 2.77410331e-30
+         8.79467609e-31 3.50665136e-30] g/cm**3
+
+        >>> len(ds.r['all', 'cell_gas_density']) == len(ds.r['all', 'particle_ones'])
+        True
+
+        """
+        self.index
+        if isinstance(deposit_field, tuple):
+            ftype, deposit_field = deposit_field[0], deposit_field[1]
+        else:
+            raise RuntimeError
+
+        if not isinstance(self.index, OctreeIndex):
+            raise NotImplementedError(
+                'The mesh deposition scheme is only implemented for oct-based datasets.')
```

Comment:
I think we should implement this for grid datasets at least before we merge this. I don't think it would be terribly hard. I can try to look into that. Not saying you need to do this.

Reply:
Hey, did you have time to work on this?

Reply:
No I haven't looked at this yet. Let's see if the tests are passing, I just restarted travis as well.